### ISSUE TITLE
v0.5: DSA pqgVer: Ensure DomainSeed is exactly N bytes

### ISSUE THREAD

**Issue by smuellerDD** [usnistgov/ACVP#494]
DSA pqgVer 2048/224 SHA224: OpenSSL and libgcrypt report a failure verdict which is flagged as wrong by the ACVP server

vsID 99 / tcID 1

**Comment by smuellerDD** [comment]
Same for DSA pqgVer 2048/256 SHA-256

vsID 101 / tcID 3

**Comment by smuellerDD** [comment]
Same for vsID 100 / tcID 4

**Comment by smuellerDD** [comment]
Am Donnerstag, 8. November 2018, 18:27:40 CET schrieb Chris Celi:

Hi Chris,

> I understand. Can you determine why your IUT expects the test case to fail?
> Otherwise I'm not terribly sure how to debug this.

For that, I looked into vsID 283, tcID 3.

The ACVP Server provides the following vectors:

```
    }, {
      "tcId" : 3,
      "p" : 
"EA7345474949B57DF2CD10333A5864B06DD8B52CA08EC13E594B71DBE02812CDCEF6AC0DD75108C1C0E590BB3A57B333505D44A95CA825EF918E688759E00DD75C5022204802A9AF7295265DE8732E30ED255655F59ED312A445A3504F5E00BEEEA87934763CD02E724E0166AE2B0870D6E89C9862F528C9F2B07141192585C2E51CDC84285BE05F02412B2DF7141C1B190400E7A93438F71D83F259EE5B242C35856E501552D6413458DE8BD505F3FF30FD4C26CA13F8442ECF3BE9D31DA8070784610E0AE5D3D8D4A8C11ABBB639C2D8C4B2E04EA02D596B83919ED5D224502127AC709566C92E290ED4DE74B3874C81BE452F9E739916486B2952A674897540D2305C133C9B48020CDE2B6502DA05BD01A756179658877AD0C545400413529997F76ED812080FC0583D3D3287B8253B14DFA9C0526E518B4745DB658F765D6BEF39E53243C8C5B3530846F550FE406548B41D6728B471CC9B40C3EAFF96A80736928D4832D89CFFAFCAB2DB2786F6245FA48543619126590574AB7FC03537",
      "q" : 
"942A6B530D7B213FD9805AECCAE1EF25AE3238AE7CCDF1C78B17E9382024FD4D",
      "domainSeed" : 
"00F993958C5C3F67E0D994A27386C928A75A233243D776E36700831DA541E6343D",
      "counter" : 126
```

Using the domainSeed, my IUT generates:

P = 
fc7e53999f474fc603894181449ed1b1e3df5b189c19b36e5da387db0436b1b6e552612a6e64faff4cd95a5d5e83287160cd4407fad6998452ed2fdfdd42ca5687b793ef11bef99b088fb57c16c829af30fc6d942b58937cb4970881dd6e47eb8f6e5b539b5e7830386ea48b01158f3d22535945a5bf104d0a34487caeafe53bd770f4e850f3930f9014b7a7d004218f0c42bf21e769171287fcfdb24ecf795df4c40b52fc3732a78237a07d6ced072101e24f2abff22f33de9416f36d7a2317fd819794930ec118f317a9d64c974f1e8d44ec76255e498a018f0f3858e5c2e3b8ce856b54c50738d7f597d32acea55d38aca6912e5abe241a910f7a6c64f94bb53637f8443d3d25ea8cc228f5b8463309b2919c8702137c5f7c385be264911df13c3d631128634d188d9a58d3635ff9f00cd4d12e7a3fbd46a0052870ea9ea76426c045a40cb101c190c49f65061c66f27d25d984dde10c25bf1b616aac009e8700acbabe3a20f97954dd30d01c99a06e9cca18702a2dda9ddaa839eed1fe19
Q = 94a0b15a945c460496123db12ec29a3a430f7cb609946701e5679624bb250abd
counter = 770

So, the values are completely different and thus the comparison will fail.


But I think I found the issue. The domainSeed offered by the ACVP server is 33 
bytes in size instead of 32 bytes! When I remove the leading 00 byte in the 
domainSeed, my IUT successfully verifies the PQ values.

Ciao
Stephan

**Comment by celic** [comment]
I am just seeing this now, sorry. I can add in some logic to ensure that these leading 00 bytes don't happen here. This is something that would be hard to verify is "working" from your end (expecting the server to not exhibit some behavior...)

I'll add some tests on our end and if the issue pops up after the fix is released please let us know. This will go out with v0.5.4 when we are ready.

**Comment by celic** [comment]
I think this is the same issue from #541. I'll have to check if the v0.5.4 server release ever made it out before the furlough.

**Comment by celic** [comment]
Would it be possible to test this again? This fix should have gone out with v0.5.4.

**Comment by smuellerDD** [comment]
I did not see such an error any more. Closing.
